app/src/main/java/com/github/se/eduverse/ui/timetable/TimeTableScreen.kt

```diff
@@ -0,0 +1,486 @@
+import com.github.se.eduverse.viewmodel.TodoListViewModel
+import java.text.SimpleDateFormat
+import java.util.Calendar
+import java.util.Locale
+
+private val blue = ButtonColors(Color(0xFF00A5FF), Color.Black, Color.LightGray, Color.Black)
+private val orange = ButtonColors(Color(0xFFFFA500), Color.Black, Color.LightGray, Color.Black)
+
+@OptIn(ExperimentalMaterial3Api::class)
+@Composable
+fun TimeTableScreen(
+    timeTableViewModel: TimeTableViewModel,
+    todoViewModel: TodoListViewModel,
+    navigationActions: NavigationActions
+) {
+  LaunchedEffect(Unit) { timeTableViewModel.getWeek() }
+
+  val context = LocalContext.current
+
+  val weeklyTable by timeTableViewModel.table.collectAsState()
+  val currentWeek by timeTableViewModel.currentWeek.collectAsState()
+  var showDialog by remember { mutableStateOf(false) }
+  var newElementType by remember { mutableStateOf(ScheduledType.EVENT) }
+
+  if (showDialog) {
+    DialogCreate(context, timeTableViewModel, todoViewModel, newElementType) { showDialog = false }
+  }
+
+  Scaffold(
+      topBar = {
+        CenterAlignedTopAppBar(
+            title = {
+              Text(
+                  "Time Table",
+                  fontWeight = FontWeight.Bold,
+                  modifier = Modifier.testTag("topBarTitle"))
+            },
+            modifier = Modifier.testTag("topBar"),
+            navigationIcon = {
+              IconButton(
+                  onClick = { navigationActions.goBack() },
+                  modifier = Modifier.testTag("backButton")) {
+                    Icon(Icons.AutoMirrored.Filled.ArrowBack, contentDescription = "Go back")
+                  }
+            },
+            colors =
+                TopAppBarDefaults.centerAlignedTopAppBarColors(containerColor = Color.Transparent))
+      },
+      bottomBar = {
+        BottomNavigationMenu({ navigationActions.navigateTo(it) }, LIST_TOP_LEVEL_DESTINATION, "")
+      }) { padding ->
+        Column(modifier = Modifier.padding(padding).fillMaxSize()) {
+
+          // Buttons to add new event/task
+          Row(
+              modifier = Modifier.fillMaxWidth().padding(15.dp),
+              horizontalArrangement = Arrangement.SpaceBetween) {
+                Button(
+                    onClick = {
+                      newElementType = ScheduledType.TASK
+                      showDialog = true
+                    },
+                    modifier = Modifier.fillMaxWidth(0.45f).testTag("addTaskButton"),
+                    colors = orange) {
+                      Text("Add task")
+                    }
+                Button(
+                    onClick = {
+                      newElementType = ScheduledType.EVENT
+                      showDialog = true
+                    },
+                    modifier = Modifier.fillMaxWidth(0.82f).testTag("addEventButton"),
+                    colors = blue) {
+                      Text("Add event")
+                    }
+              }
+
```

Comment:
Using Modifier.fillMaxWidth() with fractions like 0.45f, 0.82f, or 0.9f can cause alignment and consistency issues across different devices and screen resolutions. This can affect the display of the "Add task", "Add event" buttons, and the date and time picker fields.